pkgs/development/libraries/quickfix/default.nix

```
@@ -0,0 +1,54 @@
+{ stdenv, fetchFromGitHub, autoconf, automake, libtool }:
+
+let
+  name = "quickfix";
+  version = "1.15.1";
+in stdenv.mkDerivation rec {
+  inherit name;
+
+  nativeBuildInputs = [ autoconf automake libtool ];
+
+  src = fetchFromGitHub {
+    owner = "${name}";
+    repo = "${name}";
+    rev =  "v${version}";
+    sha256 = "1fgpwgvyw992mbiawgza34427aakn5zrik3sjld0i924a9d17qwg";
+  };
+
+  # This ugly mess is to not build or link the unittests, which do not compile with c++17
+  preConfigure = ''
+     substituteInPlace ./src/Makefile.am --replace 'LDFLAGS =-L../UnitTest++ -lUnitTest++' ' '
+     substituteInPlace ./src/Makefile.am --replace 'noinst_PROGRAMS = at ut pt' 'noinst_PROGRAMS = at pt'
```

```suggestion
     substituteInPlace ./src/Makefile.am \
       --replace 'LDFLAGS =-L../UnitTest++ -lUnitTest++' ' ' \
       --replace 'noinst_PROGRAMS = at ut pt' 'noinst_PROGRAMS = at pt'
```

Also you can do this.